devito/types/basic.py

```diff
@@ -724,9 +725,20 @@ def dimensions(self):
         """Tuple of Dimensions representing the object indices."""
         return self._dimensions
 
+    @property
+    def _eval_deriv(self):
+        return self
+
+    @property
+    def on_grid(self):
```

Comment:
all introspection properties in Devito should start with `is_` or `has_`, for homogeneity. How about `is_grid_aligned` ?

also:
* it needs a docstring
* it needs to be private